cairo/ethereum/cancun/vm/interpreter.cairo

```diff
@@ -0,0 +1,140 @@
+from ethereum.cancun.vm.runtime import get_valid_jump_destinations
+from ethereum.cancun.vm.stack import Stack, StackStruct
+
+from src.utils.dict import dict_new_empty
+from src.precompiles.precompiles import Precompiles
+
+func execute_code{
+    range_check_ptr,
+    bitwise_ptr: BitwiseBuiltin*,
+    keccak_ptr: KeccakBuiltin*,
+    poseidon_ptr: PoseidonBuiltin*,
+}(message: Message, env: Environment) -> Evm {
+    alloc_locals;
+
+    // Get valid jump destinations
+    let valid_jumpdests = get_valid_jump_destinations(message.value.code);
+    // Create empty stack
+    let (dict_start: DictAccess*) = dict_new_empty();
+    let dict_ptr = dict_start;
+    tempvar empty_stack = Stack(
+        new StackStruct(
+            dict_ptr_start=cast(dict_start, StackDictAccess*),
+            dict_ptr=cast(dict_ptr, StackDictAccess*),
+            len=0,
+        ),
+    );
+    // Create empty memory
+    let (dict_start: DictAccess*) = dict_new_empty();
+    let dict_ptr = dict_start;
+    tempvar empty_memory = Memory(
+        new MemoryStruct(
+            dict_ptr_start=cast(dict_start, Bytes1DictAccess*),
+            dict_ptr=cast(dict_ptr, Bytes1DictAccess*),
+            len=0,
+        ),
+    );
+    // Create empty Logs
+    let (empty_logs: Log*) = alloc();
+    tempvar tuple_log_struct = TupleLog(new TupleLogStruct(data=empty_logs, len=0));
+    // Create empty accounts_to_delete and touched_accounts
+    let (dict_start: DictAccess*) = dict_new_empty();
+    let dict_ptr = dict_start;
+    tempvar empty_accounts_to_delete = SetAddress(
+        new SetAddressStruct(
+            dict_ptr_start=cast(dict_start, SetAddressDictAccess*),
+            dict_ptr=cast(dict_ptr, SetAddressDictAccess*),
+        ),
+    );
+    let (dict_start: DictAccess*) = dict_new_empty();
+    let dict_ptr = dict_start;
+    tempvar empty_touched_accounts = SetAddress(
+        new SetAddressStruct(
+            dict_ptr_start=cast(dict_start, SetAddressDictAccess*),
+            dict_ptr=cast(dict_ptr, SetAddressDictAccess*),
+        ),
+    );
+
+    // Initialize EVM state
+    let evm = Evm(
+        pc=Uint(0),
+        stack=empty_stack,
+        memory=empty_memory,
+        code=message.value.code,
+        gas_left=message.value.gas,
+        env=env,
+        valid_jumpdests=valid_jumpdests,
+        logs=tuple_log_struct,
+        refund_counter=Uint(0),
+        running=TRUE,
+        message=message,
+        output=Bytes(new BytesStruct(cast(0, felt*), 0)),
+        accounts_to_delete=empty_accounts_to_delete,
+        touched_accounts=empty_touched_accounts,
+        return_data=Bytes(new BytesStruct(cast(0, felt*), 0)),
+        error=cast(0, ErrorType*),
+        accessed_addresses=message.accessed_addresses,
+        accessed_storage_keys=message.accessed_storage_keys,
+    );
+
+    // TODO: Handle precompiled contracts
+    let is_precompile = Precompiles.is_precompile(evm.message.code_address.value);
```

Comment:
remove this (from old codebase), we will make a jump table for precompile addresses (as we did for instructions), just keep the whole block as todo